tick/optim/solver/svrg.py

```diff
@@ -90,20 +108,64 @@ def __init__(self, step: float = None, epoch_size: int = None,
                              self._rand_type, step, self.seed)
 
         self.variance_reduction = variance_reduction
+        self.delayed_updates = delayed_updates
 
     @property
     def variance_reduction(self):
         return next((k for k, v in variance_reduction_methods_mapper.items()
                      if v == self._solver.get_variance_reduction()), None)
 
+    @property
+    def delayed_updates(self):
+        return next((k for k, v in delayed_updates_methods_mapper.items()
+                     if v == self._solver.get_delayed_updates()), None)
+
     @variance_reduction.setter
     def variance_reduction(self, val: str):
-
         if val not in variance_reduction_methods_mapper:
             raise ValueError(
-                'variance_reduction should be one of "{}", got "{}".'.format(
-                    ', '.join(variance_reduction_methods_mapper.keys()),
+                'variance_reduction should be one of "{}", got "{}"'.format(
+                    ', '.join(sorted(variance_reduction_methods_mapper.keys())),
                     val))
-
+        if self.model is not None:
+            if val == 'avg' and self.model._model.is_sparse():
+                raise ValueError("'avg' variance reduction cannot be used "
+                                 "with delayed updates for sparse data")
```

Comment:
I agree it would be a terrible choice but shouldn't we rather raise a warning?

Reply:
It's raising only a warning now (and unittested)